frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/StartHandler.java

```diff
@@ -209,8 +210,21 @@ public APIGatewayProxyResponseEvent handleRequest(
                     Optional.ofNullable(startRequest.previousSessionId());
             CredentialTrustLevel currentCredentialStrength =
                     startRequest.currentCredentialStrength();
-            authSessionService.addOrUpdateSessionId(
-                    previousSessionId, session.getSessionId(), currentCredentialStrength);
+            if (configurationService.isStartSessionRefactorEnabled()) {
+                authSessionService.addOrUpdateSessionId(previousSessionId, session.getSessionId());
+                AuthSessionItem authSession =
+                        authSessionService
+                                .getSession(session.getSessionId())
+                                .orElseThrow(
+                                        () ->
+                                                new AuthSessionException(
+                                                        "Auth session not found in DynamoDB"));
+                authSession.setCurrentCredentialStrength(startRequest.currentCredentialStrength());
+                authSessionService.updateSession(authSession);
+            } else {
+                authSessionService.addOrUpdateSessionId(
+                        previousSessionId, session.getSessionId(), currentCredentialStrength);
+            }
```

Comment:
Could you explain why we need to write to the table twice? I agree that `addOrUpdateSessionId` as a name doesn't encapsulate that the session properties are also updated, but it could be renamed. Perhaps eg `updateSessionAndSessionId(previousSessionId, authSession)`. And then we don't have to build a new authSession within the method - we can just pass through the session we want to update it to.

Reply:
We could rename to what you suggest, but the complication comes when `upliftRequired` also needs setting in StartHandler (and potentially more session fields in future). My plan was to first update the sessionId, then update all other session fields in one go (if possible). We may find in future that we can update sessionId and all the fields at once which would be nice, but I didn't want to assume that right now